components/api_server/src/routes/report.py

```diff
@@ -126,32 +135,29 @@ def export_report_as_pdf(report_uuid: ReportId):
 @bottle.get("/api/v3/report/<report_uuid>/json", authentication_required=True)
 def export_report_as_json(database: Database, report_uuid: ReportId):
     """Return the quality-time report, including encrypted credentials for api access to the sources."""
-    date_time = report_date_time()
-    data_model = latest_datamodel(database, date_time)
-    report = latest_report(database, data_model, report_uuid)
-
-    if report:
-        if "public_key" in bottle.request.query:
-            public_key = bottle.request.query["public_key"]
-        else:  # default to own public key
-            document = database.secrets.find_one({"name": EXPORT_FIELDS_KEYS_NAME}, {"public_key": True, "_id": False})
-            if not document:  # pragma: no feature-test-cover
-                bottle.response.status = 500
-                bottle.response.content_type = "application/json"
-                return {"error": "Cannot find the public key of this Quality-time instance."}
-            public_key = document["public_key"]
-
-        encrypt_credentials(data_model, public_key, report)
-        return report
-    bottle.response.status = 404
-    return None
+    report = latest_report(database, report_uuid)
+    if not report:
+        return report_not_found_response(report_uuid)
+    if "public_key" in bottle.request.query:
+        public_key = bottle.request.query["public_key"]
+    else:  # default to own public key
+        document = database.secrets.find_one({"name": EXPORT_FIELDS_KEYS_NAME}, {"public_key": True, "_id": False})
+        if not document:  # pragma: no feature-test-cover
+            bottle.response.status = 500
```

Comment:
```suggestion
            bottle.response.status = HTTPStatus.BAD_REQUEST
```

Reply:
It's not a bad request, the problem is that the server has no public key in the database. This shouldn't happen.

Reply:
Whoops, it was a lazy copypaste I think. Meant internal server error of course :-)